pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaAvroMessageDecoder.java

```diff
@@ -142,15 +144,22 @@ private String hex(byte[] bytes) {
     @Override
     public Boolean call() throws Exception {
       try {
-        BufferedReader reader = null;
+        URLConnection conn = url.openConnection();
+        conn.setConnectTimeout(15000);
+        conn.setReadTimeout(15000);
+        LOGGER.info("Fetching schema using url {}", url.toString());
 
-        reader = new BufferedReader(new InputStreamReader(url.openStream(), "UTF-8"));
         StringBuilder queryResp = new StringBuilder();
-        for (String respLine; (respLine = reader.readLine()) != null; ) {
-          queryResp.append(respLine);
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream(), "UTF-8"))) {
+          String line;
+          while ((line = reader.readLine()) != null) {
```

Comment:
Avoid inner assignments in loop clauses, they're pretty bad style. See https://stackoverflow.com/questions/11222121/is-it-bad-to-do-inner-assignments for example.